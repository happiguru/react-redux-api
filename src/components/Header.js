import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <header className="header-container">
    <Link className="header-link" to="/">Food Types and Recipes</Link>
  </header>
);

export default Header;
