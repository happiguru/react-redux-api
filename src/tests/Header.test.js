import React from 'react';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('Header with Link', () => {
  test('full app rendering/navigating', () => {
    render(<Header />, { wrapper: MemoryRouter });
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('Foodistics')).toBeInTheDocument;
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
