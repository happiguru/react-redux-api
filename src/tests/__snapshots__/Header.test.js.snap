// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Link should match with snapshot 1`] = `
<header>
  <div
    className="header-area "
  >
    <div
      className="main-header-area"
      id="sticky-header"
    >
      <div
        className="container"
      >
        <div
          className="row align-items-center"
        >
          <div
            className="col-xl-3 col-lg-2"
          >
            <div
              className="logo"
            >
              <a
                className="text-white"
                href="/"
              >
                Foodistics
              </a>
            </div>
          </div>
          <div
            className="col-xl-6 col-lg-7"
          >
            <div
              className="main-menu white_text  d-none d-lg-block"
            >
              <nav>
                <ul
                  id="navigation"
                >
                  <li>
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Food Types and Recipes
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
          <div
            className="col-12"
          >
            <div
              className="mobile_menu d-block d-lg-none"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
